src/Interpreter.h

```diff
@@ -0,0 +1,801 @@
+/*
+ * Souffle - A Datalog Compiler
+ * Copyright (c) 2013, 2015, Oracle and/or its affiliates. All rights reserved
+ * Licensed under the Universal Permissive License v 1.0 as shown at:
+ * - https://opensource.org/licenses/UPL
+ * - <souffle root>/licenses/SOUFFLE-UPL.txt
+ */
+
+/************************************************************************
+ *
+ * @file Interpreter.h
+ *
+ * Declares the interpreter class for executing RAM programs.
+ *
+ ***********************************************************************/
+
+#pragma once
+
+#include "InterpreterIndex.h"
+#include "RamProgram.h"
+#include "RamRelation.h"
+#include "SymbolTable.h"
+
+#include "RamStatement.h"
+
+#include <functional>
+#include <ostream>
+#include <vector>
+
+namespace souffle {
+
+class InterpreterRelation {
```

Comment:
I don't like having all the classes in this file, but maybe changing this is better left to a later refactoring effort that adds interpreter namespaces and folders for file organisation.

Reply:
The idea is to fully separate RAM from Interpreter. The RAM interpreter needs to be rewritten entirely and split up into perhaps several files.